compiler/plugins/target/ROCM/test/opt_pass_plugin/GPUHello.cpp

```diff
@@ -0,0 +1,78 @@
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/Module.h"
+#include "llvm/IR/PassManager.h"
+#include "llvm/Pass.h"
+#include "llvm/Passes/PassBuilder.h"
+#include "llvm/Passes/PassPlugin.h"
+#include "llvm/Support/FormatVariadic.h"
+#include "llvm/Support/raw_ostream.h"
+
+namespace {
+
+struct GpuHello final : llvm::PassInfoMixin<GpuHello> {
+  llvm::PreservedAnalyses run(llvm::Module &module,
+                              llvm::ModuleAnalysisManager &) {
+    bool modifiedCodeGen = runOnModule(module);
+    if (!modifiedCodeGen)
+      return llvm::PreservedAnalyses::all();
+
+    return llvm::PreservedAnalyses::all();
+  }
+  bool runOnModule(llvm::Module &module);
+  // isRequired being set to true keeps this pass from being skipped
+  // if it has the optnone LLVM attribute
+  static bool isRequired() { return true; }
+};
+
+bool GpuHello::runOnModule(llvm::Module &module) {
+  bool modifiedCodeGen = false;
+  for (llvm::Function &function : module) {
+    if (function.isIntrinsic() || function.isDeclaration())
+      continue;
+
+    llvm::StringRef functionName = function.getName();
+    if (function.getCallingConv() == llvm::CallingConv::AMDGPU_KERNEL ||
+        function.getCallingConv() == llvm::CallingConv::PTX_Kernel ||
+        functionName.contains("kernel")) {
```

Comment:
Negate this check and use an early exit (`continue`) instead. See https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code.